src/initramfs/cmd/init/pkg/platform/baremetal/baremetal.go

```diff
@@ -284,10 +308,84 @@ func (d *device) Install() error {
 		}
 
 		// Extract artifact if necessary, otherwise place at root of partition/filesystem
+		// Feels kind of janky, but going to use the suffix to denote what to do
+		switch {
+		case strings.HasSuffix(artifact, ".tar"):
+			// extract tar
+			err = untar(out)
+		case strings.HasSuffix(artifact, ".xz"):
+			// extract xz
+			// Maybe change to use gzip instead of xz to use stdlib?
```

Comment:
I like the idea of using `stdlib`. This would require some reworking of the build process, but shouldn't be difficult. Might need to make a change to the kernel as well to allow `gzip` compressed initramfs.